app/src/main/java/bg/zahov/app/ui/workout/WorkoutScreen.kt

```diff
@@ -0,0 +1,768 @@
+                                val scope = rememberCoroutineScope()
+                                var isRemoved by remember {
+                                    mutableStateOf(false)
+                                }
+
+                                val dismissState = rememberSwipeToDismissBoxState(
+                                    confirmValueChange = { value ->
+                                        if (value == SwipeToDismissBoxValue.EndToStart) {
+                                            isRemoved = true
+                                        }
+                                        true
+                                    }
+                                )
+
+                                LaunchedEffect(isRemoved) {
+                                    scope.launch {
+                                        delay(150)
+                                        if (isRemoved) {
+                                            onDeleteSet(index)
+                                        }
+                                    }
+                                }
+
+                                SwipeToDismissBox(
+                                    state = dismissState,
+                                    backgroundContent = { /* noop */ },
+                                    enableDismissFromStartToEnd = false
+                                ) {
+                                    WorkoutSet(
+                                        modifier = Modifier
+                                            .fillMaxWidth()
+                                            .animateItem(),
+                                        setIndicator = if (it.setType == SetType.DEFAULT) it.setNumber else it.setType.key,
+                                        previous = it.previousResults,
+                                        weight = if (it.set.firstMetric == null || it.set.firstMetric == 0.0) "" else it.set.firstMetric.toString(),
+                                        reps = if (it.set.secondMetric == null || it.set.secondMetric == 0) "" else it.set.secondMetric.toString(),
+                                        floatArrangement = weightValues,
+                                        onInputFieldChanged = { value, type ->
+                                            onInputFieldChanged(
+                                                index,
+                                                value,
+                                                type
+                                            )
+                                        },
+                                        onChangeSetType = { onSetTypeChange(index, it) }
+                                    )
+                                }
+                            }
+                        }
+                    }
+                }
+                Column(modifier = Modifier.padding(bottom = 16.dp)) {
+                    Button(
```

Comment:
Again extract buttons in different composable. Try to have composable fragmented as possible, group them logically. Of course do not over segregate!

Reply:
👍